parsers/python.c

```diff
@@ -300,20 +300,26 @@ static const char *skipEverything (const char *cp)
 			match = 1;
 
 		/* these checks find unicode, binary (Python 3) and raw strings */
-		if (!match && (
-			!strncasecmp(cp, "u'", 2) || !strncasecmp(cp, "u\"", 2) ||
-			!strncasecmp(cp, "r'", 2) || !strncasecmp(cp, "r\"", 2) ||
-			!strncasecmp(cp, "b'", 2) || !strncasecmp(cp, "b\"", 2)))
+		if (!match && (*cp == 'u' || *cp == 'r' || *cp == 'b' ||
+					   *cp == 'U' || *cp == 'R' || *cp == 'B'))
 		{
-			match = 1;
-			cp += 1;
-		}
-		if (!match && (
-			!strncasecmp(cp, "ur'", 3) || !strncasecmp(cp, "ur\"", 3) ||
-			!strncasecmp(cp, "br'", 3) || !strncasecmp(cp, "br\"", 3)))
-		{
-			match = 1;
-			cp += 2;
+			char c1 = *(cp + 1);
+
+			if (c1 == '\'' || c1 == '"')
+			{
+				match = 1;
+				cp += 1;
+			}
+			else if (*cp != 'r' && *cp != 'R' && (c1 == 'r' || c1 == 'R'))
```

Comment:
I'd rather move this before and simply advance `cp` (well, compute the offset rather), and then test for `'` and `"`, as this check is the same in both paths.

Something like this
```c
if (!match && (*cp == 'u' || *cp == 'r' || *cp == 'b' ||
			   *cp == 'U' || *cp == 'R' || *cp == 'B'))
{
	unsigned int i = 1;

	if (*cp != 'r' && *cp != 'R' && (cp[i] == 'r' || cp[i] == 'R'))
		i++;
	if (cp[i] == '\'' || cp[i] == '"')
	{
		match = 1;
		cp += i;
	}
}
```